xarray/convert.py

```diff
@@ -107,8 +107,12 @@ def to_iris(dataarray):
         if coord.dims:
             axis = dataarray.get_axis_num(coord.dims)
         if coord_name in dataarray.dims:
-            iris_coord = iris.coords.DimCoord(coord.values, **coord_args)
-            dim_coords.append((iris_coord, axis))
+            try:
+                iris_coord = iris.coords.DimCoord(coord.values, **coord_args)
+                dim_coords.append((iris_coord, axis))
+            except ValueError:
```

Comment:
Is this the preferred way to check for valid dimension coordinates in Iris? I guess this is probably easier than re-implementing the checks ourselves.

Reply:
It is. I thought there was a Coord.copy(...) method that did the magic of creating a coord for you of the appropriate type, but think I was imagining things.

The implementation of the checks is at https://github.com/SciTools/iris/blob/master/lib/iris/coords.py#L1720-L1735.